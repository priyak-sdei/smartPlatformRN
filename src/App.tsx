import React, { useEffect, useState } from 'react';
import {
  Appearance,
  StatusBar,
  StyleSheet,
  useColorScheme,
  View,
} from 'react-native';

import { AppNavigator } from './navigation';
import { ThemeProvider } from '@src/theme/ThemeProvider';
import { initI18n } from './i18n/i18n';
import { Provider } from 'react-redux';
import { store } from '@redux/store';
import { useAppDispatch } from '@redux/hooks';
import { setTheme } from '@redux/slices/themeSlice';
import BootSplash from 'react-native-bootsplash';
import { KeyboardProvider } from 'react-native-keyboard-controller';

const AppContent = () => {
  const dispatch = useAppDispatch();
  const isDarkMode = useColorScheme() === 'dark';

  // Listen for system theme changes and update the Redux store. This makes
  // the "Automatic Dark/Light Mode" feature fully reactive.
  useEffect(() => {
    const subscription = Appearance.addChangeListener(({ colorScheme }) => {
      dispatch(setTheme(colorScheme ?? 'light'));
    });

    return () => subscription.remove();
  }, [dispatch]);

  useEffect(() => {
    BootSplash.hide({ fade: true }).catch(err =>
      console.warn('BootSplash hide error:', err),
    );
  }, []);

  return (
    <ThemeProvider>
      <View style={styles.container} testID="app-root">
        <StatusBar barStyle={isDarkMode ? 'light-content' : 'dark-content'} />
        <AppNavigator />
      </View>
    </ThemeProvider>
  );
};

function App() {
  const [i18nReady, setI18nReady] = useState(false);

  useEffect(() => {
    initI18n().then(() => setI18nReady(true));
  }, []); // Ensure  Ensure i18n is initialized before rendering your app

  if (!i18nReady) return null; // or a splash/loading component

  return (
    <Provider store={store}>
      <KeyboardProvider>
        <AppContent />
      </KeyboardProvider>
    </Provider>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
});

export default App;
