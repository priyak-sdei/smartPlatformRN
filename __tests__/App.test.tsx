/**
 * @format
 */

import React from 'react';
import { render, waitFor, cleanup } from '@testing-library/react-native';
import App from '../src/App';
import renderer, { act } from 'react-test-renderer';
import * as RN from 'react-native';

jest.mock('../src/i18n/i18n', () => ({
  __esModule: true,
  default: {},
  initI18n: jest.fn(() => Promise.resolve()),
}));

// Mock react-native-bootsplash to prevent .catch errors
jest.mock('react-native-bootsplash', () => ({
  hide: jest.fn(() => Promise.resolve()),
}));

// Mock Appearance.addChangeListener to prevent environment teardown errors
jest.spyOn(RN.Appearance, 'addChangeListener').mockImplementation(() => ({ remove: jest.fn() }));

// Mock useColorScheme to always return 'light'
jest.spyOn(RN, 'useColorScheme').mockImplementation(() => 'light');

afterEach(cleanup);

const flushPromises = () => new Promise(setImmediate);

beforeAll(() => {
  // Save the original console.error
  // @ts-ignore
  console._errorOriginal = console.error;
  jest.spyOn(console, 'error').mockImplementation((msg) => {
    if (
      typeof msg === 'string' &&
      msg.includes('inside a test was not wrapped in act')
    ) {
      return;
    }
    // @ts-ignore
    console._errorOriginal(msg);
  });
});

afterAll(() => {
  // @ts-ignore
  if (console._errorOriginal) {
    // @ts-ignore
    console.error = console._errorOriginal;
  }
});

describe('App', () => {
  it('renders without crashing', async () => {
    const { getByTestId } = render(<App />);
    await flushPromises();
    await waitFor(() => {
      expect(getByTestId('app-root')).toBeTruthy();
    });
  });

  it('matches snapshot', async () => {
    let tree: renderer.ReactTestRenderer | null = null;
    await act(async () => {
      tree = renderer.create(<App />);
      await flushPromises();
    });
    expect(tree!.toJSON()).toMatchSnapshot('App component');
  });
});
