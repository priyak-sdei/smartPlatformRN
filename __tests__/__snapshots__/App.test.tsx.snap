// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot: App component 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
  testID="app-root"
>
  <View
    style={
      {
        "flex": 1,
      }
    }
    testID="app-navigator-root"
  />
</View>
`;
